browser/src/core/Socket.js

```diff
@@ -1294,7 +1294,7 @@ app.definitions.Socket = L.Class.extend({
 			var docUrl = url.split('?')[0];
 			this._map.options.doc = docUrl;
 			this._map.options.previousWopiSrc = this._map.options.wopiSrc; // After save-as op, we may connect to another server, then code will think that server has restarted. In this case, we don't want to reload the page (detect the file name is different).
-			this._map.options.wopiSrc = encodeURIComponent(docUrl);
+			this._map.options.wopiSrc = docUrl;
```

Comment:
I was expecting to see `encodeURIComponent()`. But, if we're sure that the issue was double-encoding and that this fixes it, then great. (Although I'm not sure how double-encoding can result in breaking the missing-encoding detection.)

Reply:
We are checking wopiSrc here: https://github.com/CollaboraOnline/online/blob/master/wsd/RequestDetails.cpp#L33. That's where the code was breaking.

Reply:
@Ashod this only proves that our protection works :)